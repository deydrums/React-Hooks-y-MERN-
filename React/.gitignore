# proyectos
/00-proyectos