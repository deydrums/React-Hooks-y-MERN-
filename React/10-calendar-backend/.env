PORT=80
DB_CNN=mongodb+srv://deydrums:7VaOymNqlPNdUn12@cluster0.id2bg.mongodb.net/CalendarApp
SECRET_JWT_SEED=Esto-Es-una-PalAbRa-S3kr2t4-j3j3j3