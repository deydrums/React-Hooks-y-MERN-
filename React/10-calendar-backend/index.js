const express = require('express');
require('dotenv').config();
//Ejecutar express
var app = express();

//Directorio publico
app.use(express.static('public'));

//Rutas
//Reescribir rutas
app.use('/api/auth', require('./routes/auth'));

//TODO: CRUD: Eventos

//Escuchar peticiones
app.listen(process.env.PORT,()=>{
    console.log(`Servidor corriendo en puerto ${process.env.PORT}`)
});