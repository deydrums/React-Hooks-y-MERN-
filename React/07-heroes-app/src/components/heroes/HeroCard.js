import React from 'react'
import { Link } from 'react-router-dom';

export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters
}) => {
    return (

        <div class="card m-2">
        <img src={`./assets/heroes/${id}.jpg`} className="img img-responsive" alt={superhero}/>
            <div class="card-body">
            <h5 class="card-title">{superhero}</h5>
            <p class="card-text">{alter_ego}</p>
            <p class="card-text">{first_appearance}</p>
            <Link to={`./hero/${id}`}>
                MÃ¡s...
            </Link>
            </div>
        </div>
    )
}
