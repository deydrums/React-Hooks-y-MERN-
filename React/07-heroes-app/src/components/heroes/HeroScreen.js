import React, { useMemo } from 'react'
import { Redirect, useHistory, useParams } from 'react-router-dom';
import { getHeroesById } from '../../selectors/getHeroById';

export const HeroScreen = () => {
    const history = useHistory();

    const {heroeId} = useParams();
    
    const hero = useMemo(() => getHeroesById(heroeId), [heroeId]);
    //const hero = getHeroesById(heroeId);
    if(!hero) {
        return <Redirect to = '/'/>;
    }
    
    const handleReturn = () =>{
        if(history.length <= 2){
            history.push('/');
        }else{
            history.goBack();
        }
    }
    const {
        superhero,
        publisher,
        alter_ego,
        first_appearance,
        characters
    } = hero;

    return (
        <div className = "container container-heroe">
            <div className = "container-img">
                <img
                    src = {`../assets/heroes/${heroeId}.jpg`}
                    alt = {superhero}
                />
            </div>
            <div className = "container-info">
                <h3 className="mt-2">{superhero}</h3>
                <h5 className="mt-2">Datos</h5>
                <ul className = "list-group list-group-flush">
                    <li className = "list-group-item"><span className="badge bg-secondary m-1">Nombre original: </span>{alter_ego}</li>
                    <li className = "list-group-item"><span className="badge bg-secondary m-1">Editorial: </span>{publisher}</li>
                    <li className = "list-group-item"><span className="badge bg-secondary m-1">Por primera vez en: </span> {first_appearance}</li>
                </ul>
                <h5 className="mt-2">Nombres</h5>
                <p className = "list-group-item"> {characters}</p>

                <button 
                    className = "btn btn-outline-primary mibutton"
                    onClick={handleReturn}
                >
                    Volver
                </button>
            </div>

        </div>
    )
}
