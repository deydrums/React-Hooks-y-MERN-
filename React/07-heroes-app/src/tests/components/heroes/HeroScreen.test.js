import React from 'react';
import {mount} from 'enzyme';
import { HeroScreen } from '../../../components/heroes/HeroScreen';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Pruebas en HeroScreen', () => {
    const history = {
        push: jest.fn(),
        length: 10,
        goBack: jest.fn(),
    }



    test('Debe de mostrar el componente redirect si no hay argumentos en el URL', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries = {['/hero']}>
                    <HeroScreen history={history}/>
            </MemoryRouter>  
        );
        expect(wrapper.find('Redirect').exists()).toBe(true);
    });

    test('Debe de mostrar un hero si el parametro existe y se encuetra', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries = {['/hero/marvel-captain']}>
                    <Route path="/hero/:heroeId" component={HeroScreen} />
            </MemoryRouter>  
        );
        expect(wrapper.find('.container-heroe').exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de regresar a la pantalla anterior con PUSH', () => {
        const history = {
            push: jest.fn(),
            length: 1,
            goBack: jest.fn(),
        }

        const wrapper = mount(
            <MemoryRouter initialEntries = {['/hero/marvel-captain']}>
                    <Route path="/hero/:heroeId" component={() => <HeroScreen history={history}/>} />
            </MemoryRouter>  
        );
    
        wrapper.find('button').prop('onClick')();
        expect(history.push).toHaveBeenCalledTimes(1);
        expect(history.push).toHaveBeenCalledWith('/');
        expect(history.goBack).not.toHaveBeenCalled();
    });

    test('Debe de regresar a la pantalla anterior', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries = {['/hero/marvel-captain']}>
                    <Route path="/hero/:heroeId" component={() => <HeroScreen history={history}/>} />
            </MemoryRouter>  
        );
    
        wrapper.find('button').prop('onClick')();
        expect(history.push).toHaveBeenCalledTimes(0);
        expect(history.push).not.toHaveBeenCalled();
        expect(history.goBack).toHaveBeenCalled();
        expect(history.goBack).toHaveBeenCalledTimes(1);
    });

    test('Debe de llamar el redirect si el hero no existe', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries = {['/hero/marvel-captain123123']}>
                    <Route path="/hero/:heroeId" component={() => <HeroScreen history={history}/>} />
            </MemoryRouter>  
        );
    
        expect(wrapper.text()).toBe('');
    });
    

})
