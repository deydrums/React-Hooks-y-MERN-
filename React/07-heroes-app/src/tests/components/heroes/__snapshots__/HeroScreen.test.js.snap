// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HeroScreen Debe de mostrar un hero si el parametro existe y se encuetra 1`] = `
<div
  className="container container-heroe"
>
  <div
    className="container-img animate__animated animate__fadeIn"
  >
    <img
      alt="Captain America"
    />
  </div>
  <div
    className="container-info"
  >
    <h3
      className="mt-2"
    >
      Captain America
    </h3>
    <h5
      className="mt-2"
    >
      Datos
    </h5>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <span
          className="badge bg-secondary m-1"
        >
          Nombre original: 
        </span>
        Steve Rogers
      </li>
      <li
        className="list-group-item"
      >
        <span
          className="badge bg-secondary m-1"
        >
          Editorial: 
        </span>
        Marvel Comics
      </li>
      <li
        className="list-group-item"
      >
        <span
          className="badge bg-secondary m-1"
        >
          Por primera vez en: 
        </span>
         
        Captain America Comics #1
      </li>
    </ul>
    <h5
      className="mt-2"
    >
      Nombres
    </h5>
    <p
      className="list-group-item"
    >
       
      Steve Rogers
    </p>
    <button
      className="btn btn-outline-primary mibutton"
      onClick={[Function]}
    >
      Volver
    </button>
  </div>
</div>
`;
